internal/core/services/room_manager/status.go

```diff
@@ -77,6 +135,40 @@ func (m *RoomManager) validateRoomStatusTransition(currentStatus game_room.GameR
 	return nil
 }
 
+// updateGameRoomStatus updates the game based on the ping and runtime status.
+func (m *RoomManager) updateGameRoomStatus(ctx context.Context, schedulerId, gameRoomId string) error {
```

Comment:
@gabrielcorado could we have a specific test for this method? i noticed that we are not covering those first four if's in room_manager tests

Reply:
Sure. Some use cases are being tested on the `UpdateRoom` test, but I'll add tests for this function.